support/map.h

```diff
@@ -2014,13 +2014,14 @@ Mono_Posix_ToUtimbuf (struct utimbuf *from, struct Mono_Posix_Utimbuf* to);
 
 
 struct Mono_Unix_UnixSignal_SignalInfo {
-	int   signum;
-	int   count;
+	int   signum;   // Read by signal handlers
+	int   count;    // Read and written by signal handlers
 	int   read_fd;
-	int   write_fd;
+	int   write_fd; // Read by signal handlers
+	int   pipecnt;  // Read by signal handlers
```

Comment:
...which brings us to ordering: the ordering (and naming) of fields here _must_ match that of `Mono.Unix.UnixSignal.SignalInfo`, in `UnixSignal.cs`.

You may also need to consider synchronization between the C# and C sides of this, e.g. C# accesses `SignalInfo.count`.